action/protocol/execution/protocol.go

```diff
@@ -37,10 +39,9 @@ func (p *Protocol) Handle(ctx context.Context, act action.Action, sm protocol.St
 
 	raCtx, ok := protocol.GetRunActionsCtx(ctx)
 	if !ok {
-		return nil, errors.New("failed to get RunActionsCtx")
+		log.S().Panic("Miss run action context")
```

Comment:
Should be consistent about missing context -- either panic everywhere or return an error. I prefer the former.